pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -923,3 +970,51 @@ func (r *Reconciler) cleanupDefaultStorageClassAddon(ctx context.Context, cluste
 	}
 	return nil
 }
+
+// addRevisionLabels adds revision labels to the given controller-runtime Object.
+func (r *Reconciler) addRevisionLabels(ctx context.Context, cluster *kubermaticv1.Cluster, obj ctrlruntimeclient.Object) (ctrlruntimeclient.Object, error) {
+	userClusterClient, err := r.kubeconfigProvider.GetClient(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	modifiedObj, err := modifier.AddRevisionLabelsToObject(ctx, userClusterClient, obj)
+	if err != nil {
+		return nil, err
+	}
+
+	return modifiedObj, nil
+}
+
+func (r *Reconciler) addCSIRevisionLabels(
+	ctx context.Context,
+	cluster *kubermaticv1.Cluster,
+	parsedUnstructuredObj *metav1unstructured.Unstructured,
+) (*metav1unstructured.Unstructured, error) {
+	if cluster == nil || cluster.Spec.Cloud.Openstack == nil ||
+		parsedUnstructuredObj == nil || parsedUnstructuredObj.GetKind() != "Deployment" ||
+		parsedUnstructuredObj.GetName() != "openstack-cinder-csi-controllerplugin" {
```

Comment:
nit: can we put the strings in conts?

Reply:
Sure! Updated in bc53dec1874ca38518c821bd18486ffb1902b634